packages/hub/node-tests/services/discord-bots/hub-discord-bots-db-gateway-test.ts

```diff
@@ -21,10 +21,9 @@ describe('HubDiscordBotsDatabaseGateway', function () {
   describe('updateStatus', function () {
     it('adds a database row if none present', async function () {
       let client = await dbManager.getClient();
-      let rows = (await client.query('SELECT * FROM discord_bots')).rows;
-      expect(rows.length).to.equal(0);
+      await client.query(`DELETE FROM discord_bots`);
```

Comment:
There was state leakage after we fixed up the DI--it looked like this test assumes the DB is clean, so i made sure by removing all the rows first.

Reply:
Is there something you could put in the bot service that automatically cleans up on tear down? The container will automatically call teardown() on the service if it's there (Example: https://github.com/cardstack/cardstack/blob/main/packages/db/index.ts#L37-L45)

I'm not sure though what the state is and whether you want it to be long lived....

Reply:
or is this something you could put in the `beforeEach` of your test helper?

Reply:
well we are--the HubBot has a `teardown()` which hooks into the DI teardown. I did explore clearing this DB table in all the bot related tests, but that did not stop the state leaking. I'm not exactly sure what test is causing this leak.

Reply:
but in general i think it's a good pattern that any test that depends on DB state needs to initialize the DB state first. Even though the container is new for each test--that doesn't change what is in the DB.

Reply:
I did try using the DB clear in a before each, and that caused the non-transactional tests in this module to blow up because the test cleanup triggered trigger pg notifications on that the non-transactional tests are sensitive to